cql3/statements/ks_prop_defs.cc

```diff
@@ -35,17 +39,21 @@ static std::map<sstring, sstring> prepare_options(
     // See issue CASSANDRA-14303.
 
     std::optional<sstring> rf;
+    bool old_rf_option = false;
     auto it = options.find(ks_prop_defs::REPLICATION_FACTOR_KEY);
     if (it != options.end()) {
         // Expand: the user explicitly provided a 'replication_factor'.
         rf = it->second;
         options.erase(it);
+        logger.warn("Using the \"replication_factor\" option is not recommended, but was used for keyspace \"{}\". "
+                "Use per-datacenter replication factor options instead.", ks_name);
```

Comment:
Warning in the log is not helpful. It will likely be noticed a long time after the keyspace was used. Better to warn via CQL.

Why is it not recommended across the board? I can understand it in some cases, but here it's too broad.